CenterForOpenScience/ember-osf-web
# [ENG-2470] Remove Contributor

- Ticket: [ENG-2470](https://openscience.atlassian.net/browse/ENG-2470)
- Feature flag: n/a

## Purpose

Allow admins on a draft registration to be able to remove contributors.

<img width="673" alt="Screen Shot 2021-01-20 at 10 44 48 AM" src="https://user-images.githubusercontent.com/6599111/105199464-c8b56b80-5b0c-11eb-9db4-e3fd122809bc.png">


## Summary of Changes

1. Add the `DeleteButton` to the editable contributor card
2. Add a `removeContributor` task to the contributor manager
3. Add localized text
4. Wire everything up
5. Add tests

## Side Effects

No, this is pretty isolated.

## QA Notes

This will be visible on the draft registration's metadata page when it's all wired in. It would be good to verify that you get the right messages from the back-end (visible in the toast error notification) when you try to delete users inappropriately.
